libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkClientImpl.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Pet peeve, and minor if it doesn't change.  But is there a better name for this?  Of course it's an "impl".  Isn't all code?  HttpBotFrameworkClient?  

Love that it's 'internal'

Reply:
Or to be consistent with another class in the same namespace: BuiltinBotFrameworkClient

Reply:
yes "Impl" is a total cop out... I was trying to be consistent with the UserTokenClientImpl (also internal implementation of abstract - basically same pattern)

"Builtin" actually meant something slightly different. It meant using the builtin constants rather than parameterized from configuration. (Again all these names are internal - though now you bring this up I'm also thinking a poor name.)

I'm more than happy to change the name(s)! suggestions more than welcome

Reply:
Now about BotFrameworkClientInternal?